ec/ecresource/deploymentresource/private_state.go

```diff
@@ -57,3 +60,38 @@ func updatePrivateStateTrafficFilters(ctx context.Context, state PrivateState, f
 
 	return state.SetKey(ctx, trafficFilterStateKey, filterBytes)
 }
+
+func readPrivateStateMigrateTemplateRequest(ctx context.Context, state PrivateState) (*deployments.MigrateDeploymentTemplateOK, diag.Diagnostics) {
```

Comment:
It makes sesne to create unit tests for the read and write funcs.

Reply:
Added in https://github.com/elastic/terraform-provider-ec/pull/755/commits/039678ffa9d5d2f37b96826199e221b749053e07.